packages/lodestar/src/db/controller/impl/level.ts

```diff
@@ -82,11 +82,19 @@ export class LevelDbController extends EventEmitter implements IDatabaseControll
   public search(opts: ISearchOptions): Promise<any> {
     return new Promise<any[]>((resolve) => {
       const searchData: any[] = [];
-      this.db.createValueStream({
+      this.db.createReadStream({
         gt: opts.gt,
         lt: opts.lt,
-      }).on("data", function (data) {
-        searchData.push(data);
+      }).on("data", function ({key, value}) {
+        // Typechecking keys as they
+        // can be defined as any type
+        if (typeof key === "number") {
```

Comment:
I think there's a tricky issue with this here.
Where this is called, in `getAllBetween`, you can see that in practice, we "encode" the keys, so here, the keys will not be numbers, and not the keys that are used within the application. They're prepended with the "bucket", an identifier of the category of the data, and are buffers.

Reply:
we can track by a new `count` variable internally so we don't have to base on type of key I think